test/e2e/network/netpol/network_policy.go

```diff
@@ -1150,6 +1153,74 @@ var _ = common.SIGDescribe("Netpol [Feature:SCTPConnectivity][LinuxOnly][Disrupt
 	})
 })
 
+var _ = common.SIGDescribe("Netpol [Feature:NetworkPolicyEndPort]", func() {
+	f := framework.NewDefaultFramework("netpol-port-range")
+
+	ginkgo.BeforeEach(func() {
+		// should open the feature gate: NetworkPolicyEndPort
+		e2eskipper.SkipUnlessNetworkPolicyEndPortEnabled()
+	})
+
+	ginkgo.Context("NetworkPolicy between server and client using port range", func() {
+		ginkgo.BeforeEach(func() {
+			_, _, _, model, k8s := getK8SModelForEndPort(f)
+			initializeResourcesByFixedNS(k8s, model)
+		})
+
+		ginkgo.AfterEach(func() {
+			if !useFixedNamespaces {
+				_, _, _, model, k8s := getK8SModelForEndPort(f)
+				framework.ExpectNoError(k8s.deleteNamespaces(model.NamespaceNames), "unable to clean up UDP netpol namespaces")
+			}
+		})
+
+		ginkgo.It("should enforce policy to allow ingress traffic using port range [Feature:NetworkPolicy]", func() {
```

Comment:
@knabben @rikatz  any other cases, we need to add ?

Reply:
Maybe in one of ingress/egress below, we can add a testcase for endPort used together with podSelector in the peer? i.e. allow ingress from pod:b on port 80-82.